pkg/condition/conditions.go

```diff
@@ -37,10 +44,35 @@ type Conditions interface {
 	IsPodSGPEnabled() bool
 }
 
-func NewControllerConditions(dataStoreSyncFlag *bool, log logr.Logger) Conditions {
+var (
+	conditionWindowsIPAMEnabled = prometheus.NewGaugeVec(
```

Comment:
we don't need a vector here since this is not multi dimensional. `prometheus.NewGauge` should be fine.